src/ChannelManager.h

```diff
@@ -0,0 +1,45 @@
+#pragma once
+
+#include <unordered_map>
+#include <set>
+
+#include "StringUtils.h"
+
+
+class cClientHandle;
+class cCriticalSection;
+class cPluginLua;
+class cChannelCallback;
+
+
+// tolua_begin
+class cChannelManager
+{
+
+public:
+	typedef SharedPtr<cChannelCallback> cChannelCallbackRef;
+	typedef std::unordered_map<AString, cChannelCallbackRef> CallbackMap;
+
+	void AddClientToChannel      (cClientHandle & a_Client, const AString & a_Channel);
+	void AddClientToChannels     (cClientHandle & a_Client, const AString & a_Channels);
+	void RemoveClientFromChannel (cClientHandle & a_Client, const AString & a_Channel);
+	void RemoveClientFromChannels(cClientHandle & a_Client, const AString & a_Channels);
+	bool ClientHasChannel        (cClientHandle & a_Client, const AString & a_Channel);
+
+	void BroadcastChannelMessage (const AString & a_Channel,  const AString & a_Data, cClientHandle & a_Client, bool a_WorldOnly = true);
```

Comment:
Originally, it was to allow for filtering, but it is no longer necessary, so I will change over to what you suggested.